From ca62452b7809b25824d3af516901504254c220fd Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Thu, 28 Nov 2024 11:01:38 +0100
Subject: [PATCH 747/841] fix(stm32mp2): fix CID filtering for EXTI2_CmIMRx

Deactivate CID filtering on EXTI2_C2IMRx and EXTI2_C3IMRx
to allow acces to this register in power off sequence and avoid IAC
with OpenSTLinux RIF configuration.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git fbce919807d4e5ca982c46f78142fcb45e2ebec3 fix(stm32mp2): fix CID filtering for EXTI2_CmIMRx]
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: If7e63fcd6636ada6ff71a00db228d4c36c749d79
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/421335
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 plat/st/stm32mp2/stm32mp2_def.h | 2 ++
 plat/st/stm32mp2/stm32mp2_pm.c  | 4 ++++
 2 files changed, 6 insertions(+)

diff --git a/plat/st/stm32mp2/stm32mp2_def.h b/plat/st/stm32mp2/stm32mp2_def.h
index b2698bc89..214420557 100644
--- a/plat/st/stm32mp2/stm32mp2_def.h
+++ b/plat/st/stm32mp2/stm32mp2_def.h
@@ -383,6 +383,8 @@ enum ddr_type {
 #define EXTI2_C3IMR3			U(0x120)
 #endif /* !STM32MP21 */
 
+#define EXTI_CmCIDCFGR(n)		(0x300U + ((n) * 4U))
+
 /*******************************************************************************
  * STM32MP2 GPIO
  ******************************************************************************/
diff --git a/plat/st/stm32mp2/stm32mp2_pm.c b/plat/st/stm32mp2/stm32mp2_pm.c
index ef426cec9..c975445ba 100644
--- a/plat/st/stm32mp2/stm32mp2_pm.c
+++ b/plat/st/stm32mp2/stm32mp2_pm.c
@@ -987,10 +987,14 @@ static void __dead2 stm32_system_off(void)
 #if !STM32MP21
 	mmio_write_32(exti2_base + EXTI2_C1IMR3, 0U);
 #endif /* !STM32MP21 */
+	/* Deactivate CID filtering on EXTI2_C2IMRx */
+	mmio_write_32(exti2_base + EXTI_CmCIDCFGR(1U), 0U);
 	mmio_write_32(exti2_base + EXTI2_C2IMR1, 0U);
 	mmio_write_32(exti2_base + EXTI2_C2IMR2, 0U);
 #if !STM32MP21
 	mmio_write_32(exti2_base + EXTI2_C2IMR3, 0U);
+	/* Deactivate CID filtering on EXTI2_C3IMRx */
+	mmio_write_32(exti2_base + EXTI_CmCIDCFGR(2U), 0U);
 	mmio_write_32(exti2_base + EXTI2_C3IMR1, 0U);
 	mmio_write_32(exti2_base + EXTI2_C3IMR2, 0U);
 	mmio_write_32(exti2_base + EXTI2_C3IMR3, 0U);
-- 
2.47.2

